static/app/views/starfish/components/chart.tsx

```diff
@@ -435,10 +435,42 @@ function Chart({
               <BarChart
                 height={height}
                 series={trimmedSeries}
-                xAxis={xAxis}
-                additionalSeries={transformedThroughput}
-                yAxes={areaChartProps.yAxes}
-                tooltip={areaChartProps.tooltip}
+                xAxis={{
+                  type: 'category',
+                  axisTick: {show: true},
+                  truncate: Infinity, // Show axis labels
+                  axisLabel: {
+                    interval: 0, // Show _all_ axis labels
+                  },
+                }}
+                yAxis={{
+                  minInterval: durationUnit ?? getDurationUnit(data),
+                  splitNumber: definedAxisTicks,
+                  max: dataMax,
+                  axisLabel: {
+                    color: theme.chartLabel,
+                    formatter(value: number) {
+                      return axisLabelFormatter(
+                        value,
+                        aggregateOutputFormat ?? aggregateOutputType(data[0].seriesName),
```

Comment:
It feels like a code smell to have so many boolean checks around with `aggregateOutputFormat ??`, `durationUnit ??` and `date` ternary, I wish we parsed data in terms of the props into a cleaner state above this.

I guess that is also a consequence of this function body being so large

Reply:
Oh yeah, this whole file is ... a little stinky!